acceptance/tests/connect/connect_inject_test.go

```diff
@@ -18,25 +18,135 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// Test that Connect works in a default and a secure installation.
+// TestConnectInject tests that Connect works in a default and a secure installation.
 func TestConnectInject(t *testing.T) {
-	cases := []struct {
+	cases := map[string]struct {
+		clusterKind consul.ClusterKind
+		releaseName string
 		secure      bool
 		autoEncrypt bool
 	}{
-		{false, false},
-		{true, false},
-		{true, true},
+		"Helm install without secure or auto-encrypt": {
+			clusterKind: consul.Helm,
+			releaseName: helpers.RandomName(),
+		},
+		"Helm install with secure": {
+			clusterKind: consul.Helm,
+			releaseName: helpers.RandomName(),
+			secure:      true,
+		},
+		"Helm install with secure and auto-encrypt": {
+			clusterKind: consul.Helm,
+			releaseName: helpers.RandomName(),
+			secure:      true,
+			autoEncrypt: true,
+		},
+		"CLI install without secure or auto-encrypt": {
+			clusterKind: consul.CLI,
+			releaseName: consul.CLIReleaseName,
+		},
+		"CLI install with secure": {
+			clusterKind: consul.CLI,
+			releaseName: consul.CLIReleaseName,
+			secure:      true,
+		},
+		"CLI install with secure and auto-encrypt": {
+			clusterKind: consul.CLI,
+			releaseName: consul.CLIReleaseName,
+			secure:      true,
+			autoEncrypt: true,
+		},
 	}
 
-	for _, c := range cases {
-		name := fmt.Sprintf("secure: %t; auto-encrypt: %t", c.secure, c.autoEncrypt)
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			cfg := suite.Config()
+			ctx := suite.Environment().DefaultContext(t)
+
+			connHelper := ConnectHelper{
+				ClusterKind: c.clusterKind,
+				Secure:      c.secure,
+				AutoEncrypt: c.autoEncrypt,
+				ReleaseName: c.releaseName,
+				T:           t,
+				Ctx:         ctx,
+				Cfg:         cfg,
+			}
+
+			connHelper.Setup()
+
+			connHelper.Install()
+			connHelper.DeployClientAndServer()
+			connHelper.TestConnectionFailureWithoutIntention()
+			connHelper.CreateIntention()
+			connHelper.TestConnectionSuccess()
+			connHelper.TestConnectionFailureWhenUnhealthy()
+		})
+	}
+}
+
+// TestConnectInjectOnUpgrade tests that Connect works before and after an
+// upgrade is performed on the cluster.
+func TestConnectInjectOnUpgrade(t *testing.T) {
+	cases := map[string]struct {
+		clusterKind         consul.ClusterKind
+		releaseName         string
+		secure, autoEncrypt bool
```

Comment:
These seem to be unused in the tests below

Reply:
Good catch